### PR TITLE
Search through subtitles

### PR DESCRIPTION
**Pull request by joschahenningsen**
### Motivation and Context
This PR implements searching through the subtitles of an individual stream, allowing students to quickly find a topic in them. 
Course-wide search could be a follow up.

Related to #796 

### Description

This PR adds
- A cron job that exports all subtitles to meilisearch once a day
- a search bar that calls a new api which internally asks meilisearch for related results
- a popup on the watch page that displays the results and allows jumping to the timestamp of the subtitle

### Steps for Testing

Prerequisites:
- 1 vod with subtitles
- a running meilisearch instance
```
docker run -it --rm \
    -p 7700:7700 \
    -e MEILI_MASTER_KEY='MASTER_KEY'\
    -v $(pwd)/meili_data:/meili_data \
    getmeili/meilisearch:v0.30 \
    meilisearch --env="development"
```
- that is configured in `config.yml`
```yaml
meili:
  host: http://localhost:7700
  apiKey: MASTER_KEY
```

1. Log in
2. Navigate to `/admin/maintenance` and run the meili export cron job
3. navigate to vod and test the search functionality 

### Screenshots

https://user-images.githubusercontent.com/44805696/222977320-f8e8ee8f-119a-4e56-8d70-82d6500c4a22.mp4

### CODE REVIEW

**Comment by mono424** [comment]
This unbelievably sexy!!!!!!

File: api/search.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ import (
 func configGinSearchRouter(router *gin.Engine, daoWrapper dao.DaoWrapper) {
 	routes := searchRoutes{daoWrapper}
 	router.GET("/api/search/streams", routes.searchStreams)
+	g := router.Group("/api/search/subtitles")
```
Comment (MatthiasReumann) [inline api/search.go]: Minor:
We could theoretically group it as `/api/search` and then `.GET("/streams")` 
and `.GET("subtitles")`. Makes the code cleaner, I guess.

File: web/ts/TUMLiveVjs.ts [hunk 2]
```diff
@@ -584,6 +584,14 @@ export function jumpTo(hours: number, minutes: number, seconds: number) {
     }
 }
 
+export function jumpToMs(ms: number) {
```
Comment (MatthiasReumann) [inline web/ts/TUMLiveVjs.ts]: Could you maybe merge the exisiting `jumpTo` function with this one by 
changing the parameter to an object? e.g.:

```typescript 
export function jumpTo(setting: object) {
  for (let j = 0; j < players.length; j++) {
         players[j].ready(() => {
           if ('ms' in object) { ... }
           else { ... }
       }
    }
}
```

**Review by MatthiasReumann — CHANGES_REQUESTED** [review]
I'm not sure how but you read my mind, I was about to suggest that feature next Wednesday! 
Works great, Looks great 💯


What's your take on this: Do you think we could already implement a TUM-Live "complete" search using Meili? e.g. for the _new_ start page.

Added some minor request changes.

File: api/search.go (hunk 1 above)
Comment (joschahenningsen) [reply api/search.go]: Done :ok: Tho I was thinking about removing the ocr search considering the quality of the subtitles, this is something we can discuss on Wednesday :)

File: web/ts/TUMLiveVjs.ts (hunk 2 above)
Comment (joschahenningsen) [reply web/ts/TUMLiveVjs.ts]: Yup, good catch :+1:

**Comment by joschahenningsen** [comment]
> Do you think we could already implement a TUM-Live "complete" search using Meili? e.g. for the new start page.

Definitely a feature I want! But currently not really feasible in my opinion until we have more subtitles. If a user had a search now they would always only find results in the one lecture with subtitles which might be confusing.

**Review by MatthiasReumann — APPROVED** [review]
Great thanks ⚡

**Pull request merged**